packaging/linux/deb/pkg.go

```diff
@@ -129,13 +134,24 @@ func BuildDeb(worker llb.State, spec *dalec.Spec, srcPkg llb.State, distroVersio
 	return dalec.MergeAtPath(llb.Scratch(), []llb.State{st, srcPkg}, "/"), nil
 }
 
-func debSources(worker llb.State, sources map[string]llb.State, opts ...llb.ConstraintsOpt) map[string]llb.State {
-	out := make(map[string]llb.State, len(sources))
+// func debSources(worker llb.State, sources map[string]llb.State, opts ...llb.ConstraintsOpt) map[string]llb.State {
+// 	out := make(map[string]llb.State, len(sources))
+// 	res := llb.Scratch().File(llb.Mkdir("/result", 0o755))
 
+// 	for key, src := range sources {
+// 		st := dalec.Tar(worker, src, "src.tar.gz", opts...)
+// 		res = res.File(llb.Mkdir("/result/"+key, 0o755)).File(
+// 			llb.Copy(st, "/", "/result/"+key))
+// 		out[key] = st
+// 	}
+
+// 	return out
+// }
+
+func singleDeb(worker llb.State, sources map[string]llb.State, opts ...llb.ConstraintsOpt) llb.State {
```

Comment:
I think it would be worthwhile to add a new function similar to `dalec.Tar` that takes a `map[string]llb.State` (or even just `map[string]Source`...) that handles all this.

Then we can mount each source in and tar it all up at once instead of creating this dependency chain like we have here.